workshops/views.py

```diff
@@ -3103,10 +3106,131 @@ def duplicates(request):
 
     return render(request, 'workshops/duplicates.html', context)
 
+# ------------------------------------------------------------
+# Views for trainees
+
+def _homework2state(homework):
+    if homework is None:
+        return 'none'
+    elif homework.state == 'n':
+        return 'not-evaluated'
+    elif homework.state == 'p':
+        return 'passed'
+    else:  # failed homework
+        return 'none'
+
+
+@login_required
+def trainee_dashboard(request):
+    swc_form = SendHomeworkForm(submit_name='swc-submit')
+    dc_form = SendHomeworkForm(submit_name='dc-submit')
+
+    # FIXME We hardcode requirement set. What's more, what we do is a bit
+    # different for each Requirement. This will results in problems when we
+    # want to switch to dynamic RequirementSet. Any ideas how to fix that?
```

Comment:
Implementing real requirement set doesn't help, because we have different business logic for each requirement.